include/wil/filesystem.h

```diff
@@ -1045,6 +1045,125 @@ namespace wil
+    return wil::unique_hfile{CreateFileW(
+        path, dwDesiredAccess, dwShareMode, &secAttributes, OPEN_EXISTING, dwFlagsAndAttributes, nullptr)};
+}
+
+/** open existing, throws on error.
+~~~
+auto handle = wil::create_file_open(filePath.c_str());
+~~~
+*/
+inline wil::unique_hfile create_file_open(PCWSTR path, DWORD dwDesiredAccess = FILE_READ_ACCESS,
+    DWORD dwShareMode = FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+    bool inheritHandle = false,
+    DWORD dwFlagsAndAttributes = FILE_ATTRIBUTE_NORMAL)
+{
+    auto handleResult = create_file_try_open(path, dwDesiredAccess, dwShareMode, inheritHandle, dwFlagsAndAttributes);
+    THROW_LAST_ERROR_IF(!handleResult.is_valid());
+    return handleResult;
+}
+
+/** create, non-throwing, returns handle and error code as a pair.
+~~~
+auto [handle, error] = wil::create_file_try_create(filePath.c_str());
+~~~
+*/
+template<create_file_create_behavior CreateDisposition = create_file_create_behavior::overwrite_existing>
+std::pair<wil::unique_hfile, DWORD> create_file_try_create(PCWSTR path, DWORD dwDesiredAccess = FILE_READ_ACCESS | FILE_WRITE_ACCESS,
+    DWORD dwShareMode = FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+    LPSECURITY_ATTRIBUTES lpSecurityAttributes = nullptr,
+    DWORD dwFlagsAndAttributes = FILE_ATTRIBUTE_NORMAL,
+    HANDLE hTemplateFile = nullptr) noexcept
+{
+    return std::pair<wil::unique_hfile, DWORD>{CreateFileW(
+        path, dwDesiredAccess, dwShareMode, lpSecurityAttributes, static_cast<DWORD>(CreateDisposition), dwFlagsAndAttributes, hTemplateFile),
+        GetLastError()};
 }
 
+/** create, throws on error.
+~~~
+auto handle = wil::create_file_create(filePath.c_str());
+~~~
+
+There are 4 create behaviors, overwrite_existing is default. To specify the others use the template parameter.
+~~~
+auto [fileHandle, error] = create_file_try_create<create_file_create_behavior::create_only_if_new>(overWriteTarget.c_str());
+~~~
+*/
+template<create_file_create_behavior CreateDisposition = create_file_create_behavior::overwrite_existing>
+wil::unique_hfile create_file_create(PCWSTR path, DWORD dwDesiredAccess = FILE_READ_ACCESS | FILE_WRITE_ACCESS,
+    DWORD dwShareMode = FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+    LPSECURITY_ATTRIBUTES lpSecurityAttributes = nullptr,
+    DWORD dwFlagsAndAttributes = FILE_ATTRIBUTE_NORMAL,
+    HANDLE hTemplateFile = nullptr)
+{
+    auto [fileHandle, error] = std::pair<wil::unique_hfile, DWORD>{CreateFileW(
+        path, dwDesiredAccess, dwShareMode, lpSecurityAttributes, static_cast<DWORD>(CreateDisposition), dwFlagsAndAttributes, hTemplateFile),
+        GetLastError()};
+    THROW_WIN32_IF(error, !fileHandle.is_valid());
+    return std::move(fileHandle);
+}
+
+}
+
+#ifndef NO_FILE_TYPE_OPERATORS
+// namespace scope
+inline bool operator!=(const FILE_ID_128& left, const FILE_ID_128& right)
+{
+    static_assert(sizeof(left) == sizeof(right), "size must match");

```

Comment:
>sizeof(left) == sizeof(right), "size must match" [](http://example.com/codeflow?start=18&length=48)

How could this possibly fail?  This method is not templated and these types are the same.  Should it be comparing the size of some field within this type?